server/game/drawcard.js

```diff
@@ -188,6 +188,20 @@ class DrawCard extends BaseCard {
         });
     }
 
+    modifyStrength(amount, applying = true) {
+        if(amount > 0) {
+            this.game.applyGameAction('increaseStrength', this, card => {
+                card._modifyStrength(amount, applying);
+            });
+        } else if(amount < 0) {
+            this.game.applyGameAction('decreaseStrength', this, card => {
```

Comment:
While using `applyGameAction` is probably the right approach (tests are failing likely due to game not being mocked), this usage is not going to be correct. Since you're only checking that the amount is less than 0 and not whether the effect is being applied or unapplied (the `applying` param), this will prevent Dany from **expiring** any STR increases. i.e.

* Song of Summer is revealed. 
* `dany.modifyStrength(1, true)` is called. 
* Dany's STR is now +1
* A new plot is revealed. Song of Summer's effect is unapplied.
* `dany.modifyStrength(-1, false)` is called.
* Dany's STR remains +1 because the effect doesn't take into consideration that it's a positive buff being unapplied rather than a negative effect being applied.

In general, I think it might be better to do this check from the effect rather than on the player. I've been trying to pull a lot of these types of checks (as well as raising events) off of `DrawCard` and `Player` in prep for #1799 and other changes. Something like the following may work better:
```javascript
modifyStrength: function(value) {
    let gameAction = value < 0 ? 'decreaseStrength' : 'increaseStrength';
    return {
        apply: function(card, context) {
            context.game.applyGameAction(gameAction, card, card => {
                card.modifyStrength(value, true);
            });
        },
        unapply: function(card, context) {
            // Use the same game action as the apply - e.g. if they were immune to the increase then we don't want the expiration decrease here to occur.
            context.game.applyGameAction(gameAction, card, card => {
                card.modifyStrength(-value, false);
            });
        },
        order: value >= 0 ? 0 : 1000
    }
}
```

(you'd also need to make similar mods to the `dynamicStrength` effect if we have any cases where STR decrease is dynamic)

I don't think this approach is perfect either (or future proof) but it keeps DrawCard a little more agnostic. You could probably get rid of the `applying` parameter by shuffling things around a bit too. It also helps in that we don't have this weird underscored private version of the method.